### PR TITLE
[Feature](group commit) Group commit block sink support pipelineX

### PR DESCRIPTION
**Pull request by Yukang-Lian**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/pipeline/pipeline_x/pipeline_x_fragment_context.cpp [hunk 1]
```diff
@@ -365,6 +366,7 @@ Status PipelineXFragmentContext::_create_data_sink(ObjectPool* pool, const TData
                                             thrift_sink.result_sink));
         break;
     }
+    case TDataSinkType::GROUP_COMMIT_OLAP_TABLE_SINK:
```
Comment (mymeiyi) [inline be/src/pipeline/pipeline_x/pipeline_x_fragment_context.cpp:369]: this sink is deprecated

File: be/src/exec/data_sink.cpp [hunk 2]
```diff
@@ -169,7 +169,7 @@ Status DataSink::create_data_sink(ObjectPool* pool, const TDataSink& thrift_sink
     case TDataSinkType::GROUP_COMMIT_BLOCK_SINK: {
         Status status = Status::OK();
         DCHECK(thrift_sink.__isset.olap_table_sink);
-        sink->reset(new vectorized::GroupCommitBlockSink(pool, row_desc, output_exprs, &status));
+        sink->reset(new vectorized::GroupCommitBlockSink(pool, row_desc, output_exprs));
         RETURN_IF_ERROR(status);
```
Comment (mymeiyi) [inline be/src/exec/data_sink.cpp:173]: this is unused, status is not changed

File: be/src/exec/data_sink.cpp [hunk 3]
```diff
@@ -326,7 +326,7 @@ Status DataSink::create_data_sink(ObjectPool* pool, const TDataSink& thrift_sink
...
```
Comment (mymeiyi) [inline be/src/exec/data_sink.cpp:330]: ditto

File: be/src/vec/sink/writer/vgroup_commit_block_writer.cpp [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    _output_tuple_desc = state->desc_tbl().get_tuple_descriptor(_tuple_desc_id);
+    if (_output_tuple_desc == nullptr) {
+        LOG(WARNING) << "unknown destination tuple descriptor, id=" << _tuple_desc_id;
+        return Status::InternalError("unknown destination tuple descriptor");
+    }
+
+    _block_convertor = std::make_unique<vectorized::OlapTableBlockConvertor>(_output_tuple_desc);
+    _block_convertor->init_autoinc_info(_schema->db_id(), _schema->table_id(),
+                                        _state->batch_size());
+    return Status::OK();
```
Comment (mymeiyi) [inline be/src/vec/sink/writer/vgroup_commit_block_writer.cpp:89]: forget to call output_vexpr_ctxs prepare and open?

Comment (Yukang-Lian) [reply be/src/vec/sink/writer/vgroup_commit_block_writer.cpp:89]: They are in AsyncWriterSink.

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

File: be/src/pipeline/exec/group_commit_block_sink_operator.h [hunk 5]
```diff
@@ -30,7 +31,7 @@ class GroupCommitBlockSinkOperatorBuilder final
     GroupCommitBlockSinkOperatorBuilder(int32_t id, DataSink* sink)
             : DataSinkOperatorBuilder(id, "GroupCommitBlockSinkOperator", sink) {}
 
-    OperatorPtr build_operator() override;
+    inline OperatorPtr build_operator() override;
```
Comment (HappenLee) [inline be/src/pipeline/exec/group_commit_block_sink_operator.h:34]: while add inline？ seems unless

Comment (Yukang-Lian) [reply be/src/pipeline/exec/group_commit_block_sink_operator.h:34]: build_operator() implemented in group_commit_block_sink_operator.h. Without inline, there will be compile error "multiple definitions".

**Comment by Yukang-Lian** [comment]
Close because #33791 finished it.

**Pull request closed without merging**